### PR TITLE
[REF] l10n_es: Refactor mod 111, 115, 303 to use tax_tags engine

### PR DESCRIPTION
**Pull request by moerradi**
This commit replaces the outdated expressions and unsigned tags in mod 111, 115, and 303 with tax_tags expressions. This update follows the removal of the old technical decision after Reportalypse. A migration script is included to ensure proper conversion of accounting history. As a result, Spanish reports are more standardized and performance is significantly enhanced due to batching in the tax_tags engine.

Community-PR: https://github.com/odoo/odoo/pull/120675
Enterprise-PR: https://github.com/odoo/enterprise/pull/40777
Upgrade-PR: https://github.com/odoo/upgrade/pull/4800

Task-Id: 3126178

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by antoine162 — APPROVED** [review]
Thanks a lot, this looks all good to me! I'd like to have a look at the upgrade PR too, if you can put a link to it in the PR description.

**Comment by moerradi** [comment]
@antoine162 I was waiting for this to be approuved then complete work on upgrade, since it looks good i'll try to finish work on Upgrade PR !
Thanks !

**Comment by qdp-odoo** [comment]
note: we're missing the upgrade branch before proceeding to the merge

File: addons/l10n_es/data/account_tax_data.xml [hunk 1]
```diff
@@ -1,300 +1,10 @@
...
-        <field name="applicability">taxes</field>
-        <field name="country_id" ref="base.es"/>
-    </record>
     <record id="mod_303_61" model="account.account.tag">
         <!--Not used anymore since Q3 2021; replaced by grids 120, 122, 123 and 124-->
         <field name="name">mod303[61]</field>
         <field name="applicability">taxes</field>
         <field name="active" eval="False"/>
         <field name="country_id" ref="base.es"/>
     </record>
```
Comment (antoine162) [inline addons/l10n_es/data/account_tax_data.xml:9]: I think this refactor would also be a good opportunity to remove this account tag once and for all :smile:

File: addons/l10n_es/data/mod111.xml [hunk 2]
```diff
@@ -0,0 +1,485 @@
...
+                            <record id="mod_111_casilla_01" model="account.report.line">
+                                <field name="name">[01] Nº of recipients</field>
+                                <field name="code">aeat_mod_111_01</field>
+                                <field name="groupby">partner_id</field>
+                                <field name="foldable" eval="True"/>
+                                <field name="expression_ids">
+                                    <record id="mod_111_casilla_01_balance" model="account.report.expression">
+                                        <field name="label">balance</field>
+                                        <field name="engine">domain</field>
+                                        <field name="formula" eval="['|', ('tax_tag_ids', '=', '+mod_111_02'), ('tax_tag_ids', '=', '+mod_111_03')]"/>
```
[the remark this answers is not in the document]
Comment (moerradi) [reply addons/l10n_es/data/mod111.xml]: fixed

**Review by antoine162 — APPROVED** [review]
LGTM :+1:

File: addons/l10n_es/data/mod303.xml [hunk 3]
```diff
@@ -0,0 +1,967 @@
...
+                                                <field name="formula">mod_303_26</field>
+                                            </record>
+                                        </field>
+                                    </record>
+                                </field>
+                            </record>
+                            <record id="mod_303_casilla_27" model="account.report.line">
+                                <field name="name">[27] Total accrual contributions</field>
+                                <field name="code">aeat_mod_303_27</field>
+                                <field name="aggregation_formula">aeat_mod_303_03.balance + aeat_mod_303_155.balance + aeat_mod_303_06.balance + aeat_mod_303_09.balance + aeat_mod_303_11.balance + aeat_mod_303_13.balance + aeat_mod_303_15.balance + aeat_mod_303_18.balance + aeat_mod_303_21.balance + aeat_mod_303_24.balance + aeat_mod_303_26.balance</field>
```
Comment (antoine162) [inline addons/l10n_es/data/mod303.xml:406]:
```suggestion
                                <field name="aggregation_formula">aeat_mod_303_152.balance + aeat_mod_303_03.balance + aeat_mod_303_155.balance + aeat_mod_303_06.balance + aeat_mod_303_09.balance + aeat_mod_303_11.balance + aeat_mod_303_13.balance + aeat_mod_303_15.balance + aeat_mod_303_158.balance + aeat_mod_303_18.balance + aeat_mod_303_21.balance + aeat_mod_303_24.balance + aeat_mod_303_26.balance</field>
```
This was added by https://github.com/odoo/enterprise/issues/39477

**Review by antoine162 — COMMENTED** [review]
Just two small things: :smile: 
1. formula [27] needs a fix, see comment below
2. it would be nice if we can include the changes of https://github.com/odoo/enterprise/pull/45619 (looks like they got lost in the refactoring)

**Review by oco-odoo — CHANGES_REQUESTED** [review]

File: addons/l10n_es/data/mod111.xml [hunk 4]
```diff
@@ -0,0 +1,485 @@
...
+                            <record id="mod_111_casilla_08" model="account.report.line">
+                                <field name="name">[08] Amount of perceptions</field>
+                                <field name="code">aeat_mod_111_08</field>
+                                <field name="groupby">account_id</field>
+                                <field name="foldable" eval="True"/>
+                                <field name="expression_ids">
+                                    <record id="mod_111_casilla_08_balance" model="account.report.expression">
+                                        <field name="label">balance</field>
+                                        <field name="engine">tax_tags</field>
+                                        <field name="formula">mod_111_08</field>
```
[the remark this answers is not in the document]
Comment (moerradi) [reply addons/l10n_es/data/mod111.xml:152]: ok, true that, on it !

[the remark this answers is not in the document]
Comment (moerradi) [reply addons/l10n_es/data/mod111.xml]: fixed !

File: addons/l10n_es/data/account_tax_data.xml [hunk 5]
```diff
@@ -1,300 +1,10 @@
...
-        <field name="applicability">taxes</field>
-        <field name="country_id" ref="base.es"/>
-    </record>
-    <record id="mod_303_60" model="account.account.tag">
-        <field name="name">mod303[60]</field>
-        <field name="applicability">taxes</field>
-        <field name="country_id" ref="base.es"/>
-    </record>
     <record id="mod_303_61" model="account.account.tag">
         <!--Not used anymore since Q3 2021; replaced by grids 120, 122, 123 and 124-->
```
[the remark this answers is not in the document]
Comment (moerradi) [reply addons/l10n_es/data/account_tax_data.xml:4]: oin it !

File: addons/l10n_es/data/mod303.xml (hunk 3 above)
Comment (moerradi) [reply addons/l10n_es/data/mod303.xml]: fixed

**Review by oco-odoo — COMMENTED** [review]
@robodoo r+

**Pull request closed without merging**